Hcal/src/Hcal/HcalSimpleDigiAndRecProducer.cxx

```diff
@@ -0,0 +1,149 @@
+    }
+  }
+  for (const auto& [barID, simhits_in_bar] : hits_by_id) {
+    ldmx::HcalHit& recHit = hcalRecHits.emplace_back();
+    double edep{};
+    double time{};
+    std::vector<double> pos{0, 0, 0};
+    for (auto hit : simhits_in_bar) {
+      edep += hit->getEdep();
+      time += hit->getTime() * edep;
+      auto hitPos{hit->getPosition()};
+      pos[0] += hitPos[0] * edep;
+      pos[1] += hitPos[1] * edep;
+      pos[2] += hitPos[2] * edep;
+    }
+    ldmx::HcalID hitID{barID};
+
+    // Position smearing
+    double mean_pe{(edep / mev_per_mip) * pe_per_mip};
+    double xpos{pos[0] / edep};
+    double ypos{pos[1] / edep};
+    double zpos{pos[2] / edep};
+    time /= edep;
+
+    auto orientation{hcalGeometry.getScintillatorOrientation(barID)};
+    double half_total_width{
+        hcalGeometry.getHalfTotalWidth(hitID.section(), hitID.layer())};
+    double scint_bar_length{hcalGeometry.getScintillatorLength(hitID)};
+
+    auto stripCenter{hcalGeometry.getStripCenterPosition(hitID)};
+    if (hitID.section() == ldmx::HcalID::HcalSection::BACK) {
+      double distance_along_bar =
+          (orientation ==
+           ldmx::HcalGeometry::ScintillatorOrientation::horizontal)
+              ? xpos
+              : ypos;
+      if (orientation ==
+          ldmx::HcalGeometry::ScintillatorOrientation::horizontal) {
+        ypos = stripCenter.y();
+        xpos += random->Gaus(0, position_resolution);
+      } else {
+        xpos = stripCenter.x();
+        ypos += random->Gaus(0, position_resolution);
+      }
+      zpos = stripCenter.z();
+      // Attenuation
+      mean_pe *= exp(1. / attenuation_length);
+      double mean_pe_close =
+          mean_pe * exp(-1. *
+                        ((half_total_width - distance_along_bar) /
+                         (scint_bar_length * 0.5)) /
+                        attenuation_length);
+      double mean_pe_far =
+          mean_pe * exp(-1. *
+                        ((half_total_width + distance_along_bar) /
+                         (scint_bar_length * 0.5)) /
+                        attenuation_length);
+      int PE_close{random->Poisson(mean_pe_close + mean_noise)};
+      int PE_far{random->Poisson(mean_pe_far + mean_noise)};
+      recHit.setPE(PE_close + PE_far);
+      recHit.setMinPE(std::min(PE_close, PE_far));
+    } else {
+      // Side HCAL, no attenuation business since single ended readout
+      int PE{random->Poisson(mean_pe + mean_noise)};
+      recHit.setPE(PE);
+      recHit.setMinPE(PE);
+      // TODO: Look into this
+      xpos = stripCenter.x();
+      ypos = stripCenter.y();
+      zpos = stripCenter.z();
+    }
+
+    recHit.setID(hitID.raw());
+    recHit.setXPos(xpos);
+    recHit.setNoise(false);
+    recHit.setYPos(ypos);
+    recHit.setZPos(zpos);
+    recHit.setTime(time);
+    recHit.setSection(hitID.section());
+    recHit.setStrip(hitID.strip());
+    recHit.setLayer(hitID.layer());
+    recHit.setEnergy(edep);
```

Comment:
Can we add the `setOrientation` function like I did in https://github.com/LDMX-Software/ldmx-sw/pull/1577/files please?